### ISSUE TITLE
Documentation for adding custom TDS Service

### ISSUE THREAD

**Issue by lesserwhirls** [Unidata/thredds#709]
For 5.0, we should add documentation regarding how to add a new service to the TDS. There are a few good examples of how to minimize (or totally eliminate) the need to add any code to be added to the THREDDS Codebase, and that would probably be a good, and easy place to start. Even a simple "Hello, World!" demo service would be nice. It could do the following:

* show Spring controller annotations
* add service link to catalog
* service could return a page with information that service developers typically want, such as dataset path on disk, dataset ID, dataset parent catalog, etc.

Note that this demo service would not ship with the TDS war file, but rather be a jar that could be dropped into WEB-INF/lib directory and "just work", much like ncSOS.

Just some ideas sparked by a conversation at ESGF. A lot of interest in adding services.

**Comment by dopplershift** [comment]
Is it possible to add tags to the threddsConfig.xml with a drop-in jar?

**Comment by lesserwhirls** [comment]
The drop-in jar could potentially look in threddsConfig.xml for things. You could always have a separate config file that lives in the thredds content root directory, too, if you need to configure things. However, that get's a bit out of control.

**Comment by dopplershift** [comment]
Right, I just looked at my Jupyter service, and I made some changes to add a `JupyterConfigBean` and then added some boilerplate to `TdsConfigMapper`. It would be *awesome* if `TdsConfigMapper` were made extensible.
